### PR TITLE
content: blog: Zephyr memory layout update

### PR DESCRIPTION
**Pull request by marekmatej**
Update the pictures to illustrate the memory utilization for a default scenario with a single CPU as well as a multi-processor case.

### CODE REVIEW

**Comment by marekmatej** [comment]
@pedrominatel @f-hollow  Guys, can you take a look?  You are the only available reviewers for the articles.

**Comment by f-hollow** [comment]
@marekmatej Sorry for the delay.

I seem to have fixed the issue with unrelated broken links in CI once and for all, please rebase your branch on `upstream/main` and see if the job `Check links in diffs` passes.

I will do the review right now.

File: content/blog/esp32-bootstrapping/index.md [hunk 1]
```diff
@@ -191,32 +191,80 @@ typedef struct esp_image_load_header {
...
+It’s important to note that different Espressif SoCs can have unique memory layouts, which can affect how resources are allocated. Therefore each SoC is using slightly different linker script.
 
-But first lets talk briefly about how the link process utilize the SRAM memory on the ESP32-S3. Note, that different Espressif SoCs have different memory layout.
 
 ### Memory utilization
 
-Bellow is the image which illustrates how the application linking is done. Highlighted are details of ROM code usage in SRAM1, and I-Cache, D-Cache placement.
-The illustration is covering an single CPU scenario and does not cover the AMP case.
+Following images illustrates how the application linking is done. Highlighted are details of ROM code usage in SRAM. I-Cache and D-Cache allocations are highlighted with red color.
+Important linker symbols are in green bubbles.
```
Comment (f-hollow) [inline content/blog/esp32-bootstrapping/index.md:204]: Edit leftovers?

File: content/blog/esp32-bootstrapping/index.md [hunk 2]
```diff
@@ -191,32 +191,80 @@ typedef struct esp_image_load_header {
...
+Important linker symbols are in green bubbles. 
+
+The images below illustrate the process of application linking, focusing on how the ROM code interacts with SRAM. Key details, such as I-Cache and D-Cache allocations, are highlighted in red for clarity.
+Important linker symbols, crucial to understanding the memory layout, are shown in green bubbles.
+These visuals are designed to make the relationships between different components and memory allocations easier to follow.
+
+
+#### ESP32-S3 use-case
+
+Here we are using the ESP32-S3 as reference platform to illustrate the memory utilization, which is **suitable as an reference for majority of newer SoCs**.
```
Comment (f-hollow) [inline content/blog/esp32-bootstrapping/index.md:213]: How about this?

```suggestion
Here we are using the ESP32-S3 to illustrate memory utilization, which is **suitable as a reference for most of the newer SoCs**.
```

File: content/blog/esp32-bootstrapping/index.md [hunk 3]
```diff
@@ -191,32 +191,80 @@ typedef struct esp_image_load_header {
...
+
+- Yellow area is the memory used by the 2nd stage bootloader and can be re-claimed by the application code run-time.
```
Comment (f-hollow) [inline content/blog/esp32-bootstrapping/index.md:215]:
```suggestion
- Yellow area is the memory used by the 2nd stage bootloader that can be re-claimed by the application code run-time.
```

File: content/blog/esp32-bootstrapping/index.md [hunk 4]
```diff
@@ -191,32 +191,80 @@ typedef struct esp_image_load_header {
...
 
+
 ***NOTE:***
 *The I-Cache allocation SRAM blocks (0,1) are set by the **`PMS_INTERNAL_SRAM_ICACHE_USAGE`** bits
 and the D-Cache allocation SRAM blocks (9,10) are set by the **`PMS_INTERNAL_SRAM_DCACHE_USAGE`** bits, both in the register `PMC_INTERNAL_SRAM_USAGE_1_REG`*
 
+
+#### ESP32 use-case
+
+Here is the memory utilization for the ESP32 platform as its memory model is significantly different from other SoCs in the ESP line.
```
Comment (f-hollow) [inline content/blog/esp32-bootstrapping/index.md:246]: _ESP line_ is something new. Let's stick to either _Espressif SoCs_ or _ESP chips_?

```suggestion
Here is the memory utilization for the ESP32 platform as its memory model is significantly different from other Espressif SoCs.
```

File: content/blog/esp32-bootstrapping/index.md [hunk 5]
```diff
@@ -191,32 +191,80 @@ typedef struct esp_image_load_header {
...
+
+Following illustration is covering an single CPU scenario.
```
Comment (f-hollow) [inline content/blog/esp32-bootstrapping/index.md:248]:
```suggestion
The following picture illustrates memory utilization in a single CPU scenario.
```

File: content/blog/esp32-bootstrapping/index.md [hunk 6]
```diff
@@ -191,32 +191,80 @@ typedef struct esp_image_load_header {
...
+
+{{< figure
+    default=true
+    src="img/esp32-zephyr-memory-default.webp"
+    alt=""
+    caption="The ESP32 'default' memory layout."
+    >}}
+
+
+Following picture illustrates the memory utilization in the multi CPU scenario.
```
Comment (f-hollow) [inline content/blog/esp32-bootstrapping/index.md:258]:
```suggestion
The following picture illustrates memory utilization in a multi-CPU scenario.
```

File: content/blog/esp32-bootstrapping/index.md [hunk 7]
```diff
@@ -347,10 +395,80 @@ I (205) heap_runtime: ESP heap runtime init at 0x3fca4f70 size 273 kB.
...
 
+
 ### MCUboot Espressif port (EP)
 
 To learn how to build the MCUboot Espressif Port, check out this [article](https://docs.mcuboot.com/readme-espressif.html)
 
+
+### AMP enabled sample code
+
+AMP builds require several images to be build and flash into a target SoC. Lets use ESP32-S3 as out test platform and demonstrate the sysbuild capabilities on the sample code that is provided by the Zephyr RTOS sources.
```
Comment (f-hollow) [inline content/blog/esp32-bootstrapping/index.md:406]:
```suggestion
AMP builds require several images to be built and flashed onto a target SoC. Let's use ESP32-S3 as our test platform and demonstrate the sysbuild capabilities on the sample code provided by the Zephyr RTOS sources.
```

File: content/blog/esp32-bootstrapping/index.md [hunk 8]
```diff
@@ -347,10 +395,80 @@ I (205) heap_runtime: ESP heap runtime init at 0x3fca4f70 size 273 kB.
...
+The sample code uses IPM (Inter-Processor Mailbox) to demonstrate simple two way communication between the PROCPU core and APPCPU core. Images for both CPU cores must be loaded using the MCUboot. Note, there is no support to run AMP while using simple boot mechanism.
```
Comment (f-hollow) [inline content/blog/esp32-bootstrapping/index.md:407]: Not clear what sample code is meant here.

```suggestion
The sample code uses IPM (Inter-Processor Mailbox) to demonstrate a simple two-way communication between the `PRO_CPU` core and `APP_CPU` core. The images for both CPU cores must be loaded using MCUboot. Note that there is no support for running AMP if the Simple Boot mechanism is used.
```

File: content/blog/esp32-bootstrapping/index.md [hunk 9]
```diff
@@ -347,10 +395,80 @@ I (205) heap_runtime: ESP heap runtime init at 0x3fca4f70 size 273 kB.
...
+
+```shell
+cd zephyrproject/zephyr
+west build -b esp32s3_devkitm/esp32s3/procpu samples/drivers/ipm/ipm_esp32/ -p --sysbuild
+west flash && west espressif monitor
+```
```
Comment (f-hollow) [inline content/blog/esp32-bootstrapping/index.md:413]: Please introduce briefly what these commands do.

File: content/blog/esp32-bootstrapping/index.md [hunk 10]
```diff
@@ -347,10 +395,80 @@ I (205) heap_runtime: ESP heap runtime init at 0x3fca4f70 size 273 kB.
...
+
+Three images has been created. Image for the PROCPU is called `ipm_esp32`. Image for the APPCPU is called `ipm_esp32_remote`. Finally MCUboot image (run by PROCPU) is called `mcuboot`.
```
Comment (f-hollow) [inline content/blog/esp32-bootstrapping/index.md:415]:
```suggestion
As a result, three images should be created:

 - `ipm_esp32` - the image for the `PRO_CPU`
 - `ipm_esp32_remote` - the image for the `APP_CPU`
 - `mcuboot` - the MCUboot image, which is run by `PRO_CPU`
```

File: content/blog/esp32-bootstrapping/index.md [hunk 11]
```diff
@@ -347,10 +395,80 @@ I (205) heap_runtime: ESP heap runtime init at 0x3fca4f70 size 273 kB.
...
+
+After flashing and connecting to a target using the serial port you should be able to observe following output on the console.
```
Comment (f-hollow) [inline content/blog/esp32-bootstrapping/index.md:417]: Is it worth providing instructions or some reminders here on how to flash and connect to the serial port? If the readers are supposed to know it already, then please ignore my question.

```suggestion
After flashing and connecting to a target using the serial port, we should be able to see the following output in the console:
```

**Review by f-hollow — COMMENTED** [review]
@marekmatej Thank you for further improving the article!

I only have a few nitpicks.

File: content/blog/esp32-bootstrapping/index.md (hunk 1 above)
Comment (marekmatej) [reply content/blog/esp32-bootstrapping/index.md:204]: yep, those are leftovers. Thx!

File: content/blog/esp32-bootstrapping/index.md (hunk 10 above)
Comment (marekmatej) [reply content/blog/esp32-bootstrapping/index.md:415]: good one

File: content/blog/esp32-bootstrapping/index.md (hunk 11 above)
Comment (marekmatej) [reply content/blog/esp32-bootstrapping/index.md:417]: the flash instructions are part of the code block describing a build

File: content/blog/esp32-bootstrapping/index.md [hunk 12]
```diff
@@ -347,11 +423,88 @@ I (205) heap_runtime: ESP heap runtime init at 0x3fca4f70 size 273 kB.
...
 
-## Feature table
+
+### AMP enabled sample code
+
+AMP builds require several images to be built and flashed onto a target SoC. Let's use ESP32-S3 as our test platform and demonstrate the sysbuild capabilities on the sample code provided by the Zephyr RTOS sources.
+The Zephyr sample code (`samples/drivers/ipm/ipm_esp32`) uses IPM (Inter-Processor Mailbox) to demonstrate a simple two-way communication between the `PRO_CPU` core and `APP_CPU` core. Images for both CPU cores must be loaded using the MCUboot.
+Note that there is no support for running AMP if the Simple Boot mechanism is used.
+
+Building and flashing a complete set of images running the IPM sample:
```
Comment (f-hollow) [inline content/blog/esp32-bootstrapping/index.md:438]: How about this:

```suggestion
Now we can build and flash a complete set of images running the IPM sample:
```

**Review by f-hollow — APPROVED** [review]
Please don't forget to implement this carefully hidden [comment](https://github.com/espressif/developer-portal/pull/356#discussion_r1888015005) and check another new suggstion.

Otherwise, LGTM!

File: content/blog/esp32-bootstrapping/index.md (hunk 12 above)
Comment (marekmatej) [reply content/blog/esp32-bootstrapping/index.md:438]: thx! fixed

Comment (marekmatej) [reply content/blog/esp32-bootstrapping/index.md]: Thanks @f-hollow, I've fixed all remaining findings.

**Pull request merged**